plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -271,4 +287,90 @@ public static HoodieTableFileSystemView getFileSystemView(
         return new HoodieTableFileSystemView(
                 tableMetadata, metaClient, metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants());
     }
+
+    public static HiveHudiPartitionInfo buildHiveHudiPartitionInfo(HudiTableHandle tableHandle, String partitionName, Partition partition)
```

Comment:
Move this to `HudiBackgroundSplitLoader` and make it private since only this class uses the class in production code?  We should follow a pattern to avoid any new util methods in `HudiUtil` unless really necessary. Only case we can make it default visibility is: the tests are hard to write to validate the logic in this method without mocks.